### DISCUSSION TITLE
Table pagination 无法翻页

### DISCUSSION THREAD

**Discussion by zeyupan888** [ant-design/ant-design#43056] [Q&A]
您好，因为一些业务需求，我们无法拿到总数，所以无法设置total。这就导致每次拿到的数据只能显示一页，但这样的话就无法实现翻页了，请问有什么解决方案吗，谢谢
<img width="225" alt="image" src="https://github.com/ant-design/ant-design/assets/71034205/8de05e3b-8f0b-483d-bb7e-8b9155eb9984">

```
 <Table columns={columns} dataSource={executions} pagination={{
                        current: page,
                        pageSize: pageSize,
                        onChange: (page, pageSize) => {
                            setPage(page);
                            setPageSize(pageSize || 10);
                        },
                    }} />
```

大概几小时前，其实是可以选择下一个和上一页的，但突然table的组件就不行了，不知道会不会是ant design的server端的code change？

**Comment by Wxh16144** [comment]
> 我们无法拿到总数，所以无法设置total

是不是可以自己计算一下 total, 每次 server 端返回输后都在此基础上 `(Math.floor({serverTotal} / {pageSize}) + 1 ) * pageSize` ?

> 不知道会不会是ant design的server端的code change？

antd 是组件, 没有 server 端

**Reply by zeyupan888** [reply]
> 是不是可以自己计算一下 total,

您好，这确实是一种解决方案，但是我们的业务需求是尽量避免去改server端的代码。
比较好奇的是，昨天我用<Table pagination... />的时候，paginator是可以显示下一页并且翻页的时候是可以拿到数据的，但是刚刚突然不行了,不知道会不会是我代码本身有改动导致的？
<img width="202" alt="image" src="https://github.com/ant-design/ant-design/assets/71034205/0c3cbe4f-b36b-47f8-8d14-f6fe2d111e85">

与此同时的话，如果不从server端拿total的话，能有什么办法在paginator显示“下一页”的button吗？网上有建议说fake一个total=99999999，但这可能会导致空页面，谢谢

**Reply by Wxh16144** [reply]
antd 今天没有发布新版本, 可以使用 https://u.ant.design/codesandbox-repro 提供一个示例代码?

**Reply by zeyupan888** [reply]
> 昨天我用的时候，paginator是可以显示下一页并且翻页的时候是可以拿到数据的

这种情况已经没法repro了，但我找到了一个临时解决方案
```
<Table columns={columns} dataSource={executions} pagination={{
                        current: page,
                        pageSize: pageSize,
                        total: executions.length + 1,
                        onChange: (page, pageSize) => {
                            setPage(page);
                            setPageSize(pageSize || 10);
                        },
                    }} />
```

这样的话就能显示下一页，但有一种edge case是，当下一页没有数据的时候，翻页的时候就会是空页(但是这个空页是没法继续再往下翻的)

**Reply by zeyupan888** [reply]
等等，这样也不行，因为每次total都是固定的导致没法继续翻下一页

**Reply by zeyupan888** [reply]
https://codesandbox.io/s/antd-reproduction-template-forked-xrwpm2

这个是sandbox, 虽然有些typescript的error。
但是大概意思是我们无法知道total的数据也想要实现翻页

**Reply by zeyupan888** [reply]
ok
这样就可以了
```
<Table columns={columns} dataSource={entities} pagination={{
            current: page,
            pageSize: pageSize,
            total: page * pageSize + 1,
            onChange: (page, pageSize) => {
                setPage(page);
                setPageSize(pageSize || 10);
            },
        }} />
```
